cypress/integration/DisplaysButtonWhenUserIsAuthenticated.feature.js

```diff
@@ -0,0 +1,60 @@
+describe('Add to order button', () => {
+  describe('is visible for authenticated users', () => {
+    beforeEach(() => {
+      cy.server()
+      cy.route({
+        method: "GET",
+        url: "http://localhost:3000/api/products",
+        response: 'fixture:all_products.json',
+      });
+      cy.route({
+        method: "POST",
+        url: "http://localhost:3000/api/auth",
+        response: "fixture:register_user.json",
+        headers: {
+          uid: 'user@email.com'
+        }
+      });
+      cy.visit('/')
+    })
+
+    it('displays a form to register', () => {
+      cy.get('[data-cy="registration-form"]').within(() => {
+        cy.get('[data-cy="email-field"]').type('user@email.com')
+        cy.get('[data-cy="password-field"]').type('password')
+        cy.get('[data-cy="password-confirmation-field"]').type('password')
+        cy.get('[data-cy="submit"]').click()
+      })
+      cy.get('[cy-data="product_id_1"]').within(() => {
+        cy.get('[data-cy="order-button"]').should('be.visible')
+      })
+
+    })
+
+  })
+  describe('button not visible if user is not authenticated', () => {
+    beforeEach(() => {
+      cy.server()
+      cy.route({
+        method: "GET",
+        url: "http://localhost:3000/api/products",
+        response: 'fixture:all_products.json',
+      });
+      cy.route({
+        method: "POST",
+        url: "http://localhost:3000/api/products",
+        response: {
+          errors: "Something went wrong! Please try again!",
+          success: false
+        },
+        status: 401
+      })
+      cy.visit('/')
+    })
+    it('it is not visible', () => {
+      cy.get('[cy-data="product_id_1"]').within(() => {
+        cy.get('[data-cy="order-button"]').should('not.be.visible')
```

Comment:
I'm not really sure why you are stubbing out the request to sign in here when you actually don't perform that call to the api. 
Please make sure that you log in with the wrong credentials and then see the error message and still no button